src/hooks/useSearchState.ts

```diff
@@ -0,0 +1,44 @@
+import {NavigationRouteContext} from '@react-navigation/native';
+import {useContext, useMemo} from 'react';
+import type {SearchQueryJSON} from '@components/Search/types';
+import {buildSearchQueryJSON} from '@libs/SearchQueryUtils';
+import CONST from '@src/CONST';
+import SCREENS from '@src/SCREENS';
+
+type SearchRouteParams = {
+    q?: string;
+    type?: string;
+    hashKey?: number;
+};
+
+type SearchStateResult = {
+    isOnSearch: boolean;
+    hashKey?: number;
+};
+
+/**
+ * Hook to manage search state based on route parameters
+ * Returns search status and hash for query tracking
+ */
+const useSearchState = (): SearchStateResult => {
+    // We are using these contexts directly instead of useRoute, because those will throw an error if used outside a navigator.
+    const route = useContext(NavigationRouteContext);
+    const {q, type, hashKey: hashKeyFromRoute} = (route?.params as SearchRouteParams) ?? {};
+
+    return useMemo(() => {
+        const isSearchAttachmentModal = route?.name === SCREENS.ATTACHMENTS && type === CONST.ATTACHMENT_TYPE.SEARCH;
```

Comment:
NAB we can move this below the early return

Reply:
done